### PR TITLE
Automattic for Agencies: Add Horizon env file and update other envs

### PR DESCRIPTION
**Pull request by yashwin**
## Proposed Changes

This PR:

- Adds the Horizon env file
- Updates theme_color on all the other env files
- Updates favicon_url & enables all sections on the Staging env file.

## Testing Instructions

- Verify all the changes look good.

## Pre-merge Checklist

- [x] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [x] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

### CODE REVIEW

File: config/a8c-for-agencies-horizon.json
Comment (yashwin) [inline config/a8c-for-agencies-horizon.json:1]: Keeping this file similar to `jetpack-cloud-horizon`

File: config/a8c-for-agencies-stage.json [hunk 1]
```diff
@@ -14,11 +14,16 @@
 	},
 	"enable_all_sections": false,
 	"sections": {
-		"a8c-for-agencies": true
+		"a8c-for-agencies": true,
+		"a8c-for-agencies-auth": true,
+		"a8c-for-agencies-overview": true,
+		"a8c-for-agencies-plugins": true,
+		"a8c-for-agencies-sites": true,
+		"a8c-for-agencies-marketplace": true
```
Comment (yashwin) [inline config/a8c-for-agencies-stage.json:22]: Enabling these sections to easily track the progress for all the shareholders.

File: config/a8c-for-agencies-stage.json [hunk 2]
```diff
@@ -1,7 +1,7 @@
 {
 	"env": "production",
 	"env_id": "a8c-for-agencies-stage",
-	"favicon_url": "/calypso/images/favicons/favicon-a4a-prod.ico",
+	"favicon_url": "/calypso/images/favicons/favicon-a4a-stage.ico",
```
Comment (yashwin) [inline config/a8c-for-agencies-stage.json:4]: Fix the staging favicon

**Review by jkguidaven — APPROVED** [review]
Changes LGTM 👍

**Pull request merged**